vhostuser/vhostuser.go

```diff
@@ -120,35 +119,41 @@ func createVhostPort(conf *NetConf, containerID string) error {
 	sockPath := filepath.Join(sockDir, sockRef)
 
 	// vppctl create vhost socket /tmp/sock0 server
-	args := []string{"create", sockPath}
-	if output, err := ExecCommand(conf.VhostConf.Vhosttool, args); err == nil {
+	cmd_args := []string{"create", sockPath}
+	if output, err := ExecCommand(conf.VhostConf.Vhosttool, cmd_args); err == nil {
 		vhostName := strings.Replace(string(output), "\n", "", -1)
 
-		args = []string{"getmac", vhostName}
-		if output, err := ExecCommand(conf.VhostConf.Vhosttool, args); err == nil {
+		cmd_args = []string{"getmac", vhostName}
+		if output, err := ExecCommand(conf.VhostConf.Vhosttool, cmd_args); err == nil {
 			conf.VhostConf.VhostMac = strings.Replace(string(output), "\n", "", -1)
 		}
 
 		conf.VhostConf.Vhostname = vhostName
-		conf.VhostConf.Ifname = conf.If0name
+		conf.VhostConf.Ifname = args.IfName
 		conf.VhostConf.IfMac = GenerateRandomMacAddress()
-		return saveVhostConf(conf, containerID)
+		return saveVhostConf(conf, args)
 	}
 
 	return nil
 }
 
-func destroyVhostPort(conf *NetConf, containerID string) error {
+func destroyVhostPort(conf *NetConf, args *skel.CmdArgs) error {
 	vc := &VhostConf{}
-	if err := vc.loadVhostConf(conf, containerID); err != nil {
+	if err := vc.loadVhostConf(conf, args); err != nil {
 		return err
 	}
 
 	//vppctl delete vhost-user VirtualEthernet0/0/0
-	args := []string{"delete", vc.Vhostname}
-	if _, err := ExecCommand(conf.VhostConf.Vhosttool, args); err == nil {
-		path := filepath.Join(conf.CNIDir, containerID)
-		return os.RemoveAll(path)
+	cmd_args := []string{"delete", vc.Vhostname}
+	if _, err := ExecCommand(conf.VhostConf.Vhosttool, cmd_args); err == nil {
+		path := filepath.Join(conf.CNIDir, args.ContainerID)
+		fileName := fmt.Sprintf("%s-%s", args.ContainerID[:12], args.IfName)
+		suffixes := []string{"", ".json", "-ip4.conf"}
+		for _, suffix := range suffixes {
+			file := filepath.Join(path, fileName + suffix)
+			os.Remove(file)
+		}
+		os.Remove(path) // Fails if dir not empty
```

Comment:
Should check if dir is empty before trying to remove, and then print an warning message if removing of dir fails.